colorpicker/src/main/java/org/catrobat/paintroid/colorpicker/RgbSelectorView.kt

```diff
@@ -0,0 +1,216 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.paintroid.colorpicker
+
+import android.content.Context
+import android.graphics.Color
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.AttributeSet
+import android.view.View
+import android.widget.SeekBar
+import android.widget.SeekBar.OnSeekBarChangeListener
+import androidx.appcompat.widget.AppCompatEditText
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.appcompat.widget.LinearLayoutCompat
+import androidx.core.content.ContextCompat
+import java.lang.IllegalArgumentException
+import java.util.Locale
+import kotlin.math.min
+
+private const val CONSTANT_9 = 9
+private const val CONSTANT_2POINT55 = 2.55f
+
+class RgbSelectorView : LinearLayoutCompat {
+    private var seekBarRed: SeekBar
+    private var seekBarGreen: SeekBar
+    private var seekBarBlue: SeekBar
+    private var seekBarAlpha: SeekBar
+    private var textViewRed: AppCompatTextView
+    private var textViewGreen: AppCompatTextView
+    private var textViewBlue: AppCompatTextView
+    private var textViewAlpha: AppCompatTextView
+    private var editTextHex: AppCompatEditText
+    private var onColorChangedListener: OnColorChangedListener? = null
+
+    var selectedColor: Int
+        get() = Color.argb(
+            seekBarAlpha.progress,
+            seekBarRed.progress,
+            seekBarGreen.progress,
+            seekBarBlue.progress
+        )
+        set(color) {
+            val colorRed = Color.red(color)
+            val colorGreen = Color.green(color)
+            val colorBlue = Color.blue(color)
+            val colorAlpha = Color.alpha(color)
+            seekBarAlpha.progress = colorAlpha
+            seekBarRed.progress = colorRed
+            seekBarGreen.progress = colorGreen
+            seekBarBlue.progress = colorBlue
+            var currentCursorPosition = editTextHex.selectionStart
+            editTextHex.tag = "changed programmatically"
+            editTextHex.setText(
+                String.format(
+                    "#%02X%02X%02X%02X",
+                    colorAlpha,
+                    colorRed,
+                    colorGreen,
+                    colorBlue
+                )
+            )
+            editTextHex.tag = null
+            val editTextHexLength = editTextHex.text.toString().length
+            currentCursorPosition = min(currentCursorPosition, editTextHexLength)
+            editTextHex.setSelection(currentCursorPosition)
+            setSelectedColorText(color)
+        }
+
+    constructor(context: Context) : super(context)
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
+
+    init {
+        val rgbView = inflate(context, R.layout.color_picker_layout_rgbview, null)
+        addView(rgbView)
+        seekBarRed = rgbView.findViewById<View>(R.id.color_picker_color_rgb_seekbar_red) as SeekBar
```

Comment:
you could probably use with(rgbView or inflate(...)) {}
if you want :)

Reply:
hmm yeah nice idea 👍🏻